### PR TITLE
proposed changes

### PR DESCRIPTION
**Pull request by jonstorer**
I made some changes to the existing applicant project that force situations in order to see how the applicant will handle each.

## Objectives

1. When I load the app I see a list of businesses that includes the Business Name, Category, Number of Campaigns, and Average Campaign Budget
2. When I click on a column header, the list of businesses sorts
3. When I enter text in a search box, the list of businesses filters by name
4. When I click a business's name, I am on a new page.
   1. At the top, I see the business's name
   1. I see the business's details
   2. I see the business's location on a map (google maps or any other map utility)
   3. I see a list of the business's campaigns that includes the Name and Budget
      1. When I click on the column header, the list of campaigns sorts
      2. When I enter text in a search box, the list of campaigns filters by name

##### json files
- Is the loaded from the json files and preprocessed?
- Is it reformatted for use?

##### "List" Businesses endpoint?
- Supports sorting?
- Supports filtering?
- Supports pagination?
- Data for the list of businesses or business details or both?
- Has the data been restructured for api presentation (or is it a "database dump")?
- Where is the "average campaign budget" calcualted?
- Is sensative data scrubbed from api responses?

##### Is there a "Show" Business endpoint?
- Is the data in the show endpoint different than a single record from the list endpoint?
- Is the data in the "show" endpoint an augmentation of the data in the "list" endpoint for a single record?
- Has the data been restructured for api presentation (or is it a "database dump")?
- Where is the "average campaign budget" calcualted?
- Is sensative data scrubbed from api responses?

##### list of businesses
- Does the submission show all the required data?
- Does the submission use a table to show businesses? (Or other appropriate structure)
- Where is the average campaign budget calcualted?
- Is the long list of businesses paginated ? (Or another appropriate long list strategy)
- Does the first click of a header sort the data? Does the 2nd reverse the sort?
- Does filtering the list via the search box filter data well? In the front end? Sent to the sever? Is there a delay sending to the server?
- Does clicking a business name navigate to a new url (ex business/:id)?

##### details of a business

- Does the submission use routing to render a different view for the details page?
- Does the submission show all the required data?
- Is the calculation of the Average Campaign Budget reused (shared in the fe or be) or is the calculation duplicated? 
- Does the submission use an easy map library (component?) to show the location?
- Does the table sorting / filtering / rendering use components to share the logic between the list of businesses and campaigns?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,26 +4,33 @@
...
+
+## Technical Requirements
+1. Backend should be built with Express.js
+2. Frontend should be built with Vue.js
+
+## User Stories
+1. When I load the app I see a list of businesses that includes the Business Name, Category, Number of Campaigns, and Average Campaign Budget
+2. When I click on a column header, the list of businesses sorts
+3. When I enter text in a search box, the list of businesses filters by name
+4. When I click a business's name, I am take to a new page.
```
Comment (dpowers-advisr) [inline README.md:21]: 4. When I click a business's name, I am redirected to a new page.

File: README.md [hunk 2]
```diff
@@ -4,26 +4,33 @@
...
+   1. At the top, I see the business's name 
+   1. I see the business's details
+   2. I see the business's location on a map (google maps or any other map utility)
+   3. I see a list of the business's campaigns that includes the Name and Budget
+      1. When I click on the column header, the list of campaigns sorts
+      2. When I enter text in a search box, the list of campaigns filters by name
+
+
+## Good luck and please let us know if you have any questions!
+We will schedule time for you to come in and talk us through your completed project.
```
Comment (dpowers-advisr) [inline README.md:31]: We will schedule time for you to talk us through your completed project.

**Review by dpowers-advisr — CHANGES_REQUESTED** [review]
Just some wording things and a typo

**Review by kusala-w — APPROVED** [review]
Looks good

File: README.md (hunk 1 above)
Comment (jonstorer) [reply README.md:21]: @dpowers-advisr good call out. I'm going update this to be at least correct english. However, I'm trying to be soft in my language here. I don't want to use to leading of language to push them into using routing. I want them to extract the requirement from the "good design patterns" they already have. That said, does this language suffice?

```
4. When I click a business's name, I am taken to a new page.
```

Comment (dpowers-advisr) [reply README.md]: Yup looks good to me

**Review by dpowers-advisr — APPROVED** [review]
🚀

**Comment by JaimeNufio** [comment]
Looks good 👍

**Pull request closed without merging**